actionview/lib/action_view/helpers/form_options_helper.rb

```diff
@@ -566,9 +566,10 @@ def grouped_options_for_select(grouped_options, selected_key = nil, options = {}
       # an ActiveSupport::TimeZone.
       #
       # By default, +model+ is the ActiveSupport::TimeZone constant (which can
-      # be obtained in Active Record as a value object). The only requirement
-      # is that the +model+ parameter be an object that responds to +all+, and
-      # returns an array of objects that represent time zones.
+      # be obtained in Active Record as a value object). The +model+ parameter
+      # must respond to +all+ and return an array of objects that represent time
+      # zones; each object must respond to +name+. If a Regexp is given the
+      # object must redefine +=~+.
```

Comment:
I find this is a touch confusing, because say if an object already defines `=~` correctly through their parent, then there is no need to redefine the operator. 

A better way to express this might be: `If a Regexp is given it will attempt to match the zones using the ~= operator.`.

Reply:
You have a subtle point. Updated!